cosmwasm/register-p2p-tokens.ts

```diff
@@ -0,0 +1,219 @@
+    const chainsToProcess = new Set(chains);
+    const tokenInfoString = fs.readFileSync(`axelar-chains-config/info/tokens-p2p/tokens-${env}.json`, 'utf8');
+    const tokenInfo = JSON.parse(tokenInfoString) as SquidTokenInfoFile;
+    const interchainTokenServiceAddress = config.getContractConfig('InterchainTokenService').address;
+
+    if (!interchainTokenServiceAddress) {
+        throw new Error('InterchainTokenService contract address not found');
+    }
+
+    const promises = Object.values(tokenInfo.tokens)
+        .filter((tokenData: SquidToken) => (tokenIds ? tokenIdsToProcess.has(tokenData.tokenId) : true))
+        .flatMap((tokenData: SquidToken) => {
+            return tokenData.chains
+                .filter((chain: SquidTokenData) => {
+                    try {
+                        return (
+                            tokenData.tokenType === 'interchain' &&
+                            (chains ? chainsToProcess.has(chain.axelarChainId.toLowerCase()) : true) &&
+                            chain.axelarChainId !== tokenData.originAxelarChainId &&
+                            (chain.registered ? !chain.registered : true) &&
+                            isConsensusChain(config.getChainConfig(chain.axelarChainId.toLowerCase()))
+                        );
+                    } catch (e) {
+                        printError(`Error getting chain config for ${chain.axelarChainId} (skipping chain): ${e}`);
+                        return false;
+                    }
+                })
+                .map(async (tokenOnChain: SquidTokenData) => {
+                    return processToken(tokenData, tokenOnChain);
+                });
+        });
+    await Promise.all(promises);
+    fs.writeFileSync(`axelar-chains-config/info/tokens-p2p/tokens-${env}.json`, JSON.stringify(tokenInfo, null, 2));
+}
+
+async function registerTokensInFile(client: ClientManager, config: ConfigManager, options, _args, _fee) {
+    const interchainTokenServiceAddress = config.getContractConfig('InterchainTokenService').address;
+
+    if (!interchainTokenServiceAddress) {
+        throw new Error('InterchainTokenService contract address not found');
+    }
+
+    await forEachTokenInFile(config, options, async (tokenData: SquidToken, tokenOnChain: SquidTokenData) => {
+        try {
+            const tokenDataToRegister = {
+                tokenId: tokenData.tokenId,
+                originChain: tokenData.originAxelarChainId || getOriginChain(tokenData),
+                decimals: tokenData.decimals,
+                supply: tokenOnChain.trackSupply
+                    ? await getSupply(tokenOnChain.tokenAddress, config.getChainConfig(tokenOnChain.axelarChainId.toLowerCase()).rpc)
+                    : undefined,
+                axelarId: tokenOnChain.axelarChainId,
+            } as TokenDataToRegister;
+            await registerToken(interchainTokenServiceAddress, client, tokenDataToRegister, options.dryRun);
```

Comment:
registerToken should check that the token data is already registered and act as a no-op if so